build-scan-shared/src/buildTool/common.ts

```diff
@@ -95,6 +79,26 @@ export abstract class BuildTool {
     protected getPublisherProjectDir(): string {
         return path.resolve(this.getBuildScanWorkDir(), this.PUBLISHER_PROJECT_DIR)
     }
+}
+
+export abstract class PostPublishingBuildTool extends BuildTool {
+    private readonly SCAN_FILENAME = `scan.scan`
+
+    protected abstract getCommand(): string
+
+    protected abstract getPluginDescriptorFileName(): string
+
+    protected abstract getPluginDescriptorTemplate(): string
+
+    protected abstract getPublishTask(): string[]
+
+    createPublisherProjectStructure(): void {}
+
+    createPluginDescriptorFileWithCurrentVersion(version: string): void {
+        const resolvedContent: string = this.getPluginDescriptorTemplate().replace(this.REPLACE_ME_TOKEN, version)
```

Comment:
I think that the type is redundant here as it is clear from the context.

```suggestion
        const resolvedContent = this.getPluginDescriptorTemplate().replace(this.REPLACE_ME_TOKEN, version)
```